src/TwigYard/Middleware/Form/Handler/ApiHandler.php

```diff
@@ -0,0 +1,354 @@
+                                self::CONFIG_DATA,
+                                $dataKey,
+                                self::CONFIG_DATA_FORM_VALUE,
+                                self::CONFIG_DATA_FORM_VALUE,
+                                self::CONFIG_DATA_DEFAULT
+                            )
+                        );
+                    }
+                } elseif (!is_string($dataValue) && !is_numeric($dataValue)) {
+                    throw new InvalidSiteConfigException(
+                        sprintf(
+                            'Form API handler option `%s.%s` is not set or is not of type string or int.',
+                            self::CONFIG_DATA,
+                            $dataKey
+                        )
+                    );
+                }
+            }
+        } else {
+            throw new InvalidSiteConfigException(
+                sprintf(
+                    'Form API handler option `%s` is not set or is not of type array.',
+                    self::CONFIG_DATA
+                )
+            );
+        }
+    }
+
+    /**
+     * @throws InvalidSiteConfigException
+     */
+    private function validateConfigHeaders()
+    {
+        if (array_key_exists(self::CONFIG_HEADERS, $this->config)) {
+            $configHeaders = $this->config[self::CONFIG_HEADERS];
+
+            if (is_array($configHeaders)) {
+                foreach ($configHeaders as $headerKey => $headerValue) {
+                    if (!is_string($headerValue) && !is_numeric($headerValue)) {
+                        throw new InvalidSiteConfigException(
+                            sprintf(
+                                'Form API handler option `%s.%s` is not set or is not of types string or int.',
+                                self::CONFIG_HEADERS,
+                                $headerKey
+                            )
+                        );
+                    }
+                }
+            } else {
+                throw new InvalidSiteConfigException(
+                    sprintf(
+                        'Form API handler option `%s` is not set or is not of type array.',
+                        self::CONFIG_HEADERS
+                    )
+                );
+            }
+        }
+    }
+
+    /**
+     * @throws InvalidSiteConfigException
+     */
+    private function validateConfigResponse(): void
+    {
+        if (array_key_exists(self::CONFIG_RESPONSE, $this->config)) {
+            $configResponse = $this->config[self::CONFIG_RESPONSE];
+
+            if (is_array($configResponse)) {
+                if (
+                    count($configResponse) === 1
+                    && array_key_exists(self::CONFIG_RESPONSE_REDIRECT_URL_PARAM, $configResponse)
+                ) {
+                    $redirectUrlParam = $configResponse[self::CONFIG_RESPONSE_REDIRECT_URL_PARAM];
+
+                    if (!is_string($redirectUrlParam) && !is_numeric($redirectUrlParam)) {
```

Comment:
Please remove is_numeric validation

Reply:
This is correct. You can receive reponse `{1:'https://www.example.com'}` with numeric key. In that case following configuration is valid: 
```
    - type: api
      response:
          redirect_url_param: 1
```
It is same as in data part. Am I right?

Reply:
I was confused by error message "... is not set or is not of type string".

Reply:
Message has been fixed